spec/moonshot/stack_spec.rb

```diff
@@ -88,10 +90,32 @@
 
       it 'should call CreateStack, then wait for completion' do
         config.additional_tag = 'ah_stage'
+        expect(s3_client).not_to receive(:put_object)
         expect(cf_client).to receive(:create_stack)
           .with(hash_including(expected_create_stack_options))
         subject.create
       end
+
+      context 'when template_s3_bucket is set' do
+        before(:each) do
+          config.template_s3_bucket = 'rspec-bucket'
+        end
+
+        let(:expected_put_object_options) do
+          {
+            bucket: config.template_s3_bucket,
+            key: an_instance_of(String),
```

Comment:
If we mock out some of the components used in the key, we can assert the behavior here instead of just expecting an instance of a string (e.g. Time).